### PR TITLE
Solution day03 article

### PR DESCRIPTION
**Pull request by iusildra**
@bishabosha
I wrote the article as you suggested on my pr :slightly_smiling_face:

I tried to explain as much as possible, but if you think it needs some more explanations, I'll add them :grin:

### CODE REVIEW

File: docs/2023/puzzles/day03.md [hunk 1]
```diff
@@ -11,187 +11,184 @@ https://adventofcode.com/2023/day/3
...
+    else true
+case class PartNumber(value: Int, start: Coord, end: Coord)
+case class Symbol(sym: String, pos: Coord):
+  def neighborOf(number: PartNumber) = pos.within(
+    Coord(number.start.x - 1, number.start.y - 1),
+    Coord(number.end.x + 1, number.end.y + 1)
+  )
+
+object IsInt:
+  def unapply(in: Match): Option[Int] = in.matched.toIntOption
```
Comment (bishabosha) [inline docs/2023/puzzles/day03.md:155]: your code is missing the `import scala.util.matching.Regex.Match`

**Comment by bishabosha** [comment]
> With my solution it gives the same wrong result, and the output was the same as in `input/day03`

I'm still a bit confused what you mean - I copied your code into the repl and it gives the right output when using `input/day03`, so when is it wrong?

**Comment by iusildra** [comment]
It's wrong when I run it from the website (after I updated the solution in the `solution` submodule)
image

**Comment by iusildra** [comment]
Oh, well it seems that my copy/paste was the culprit :skull:

**Pull request merged**